lib/components/channel_localization.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+class ChannelLocalization extends ChangeNotifier {
```

Comment:
This entire class is not really necessary since you already have the property on `UserModel`. Listen on `UserModel` instead.